### ISSUE TITLE
0 BTC change output causes transaction to be rejected

### ISSUE THREAD

**Issue by niser32** [spesmilo/electrum#1839]
I was trying to send all the balance from my wallet (multiple inputs on various addresses) to a single address. I manually specified the fee and send amount to match exactly the balance on my wallet so there would be no need for a change output. But after clicking send Electrum created a transaction with 2 outputs, one was the address I was sending to with the correct amount and the other was 0,00 BTC to a change address on the same wallet.
It let me sign the transaction but when I tried to broadcast it showed an error:
error: The transaction was rejected by network rules. (64: dust)

I tried changing the fee, using the default fee and disabling change addresses but it always creates that unnecessary 0 BTC output that gets the transaction rejected.

I'm using Electrum 2.6.4 on Linux.

**Comment by ecdsa** [comment]
I cannot reproduce this on git master.
can you clarify how you did it?

**Comment by niser32** [comment]
1. Open HD wallet with balance on more than one address (or a single address with at least 2 inputs).
2. Click send tab.
3. Enter one address in the pay to field.
4. Put ! in the amount field to select the maximum amount of BTC.
5. Click Send...
Electrum now displays a transaction (I have selected "View transaction before signing" in preferences) using all the unspent inputs on the wallet but it shows 2 outputs where there should be only one. That extra output is returning 0 btc to a change address. This causes the transaction to be rejected by the network.

This makes it impossible to empty a wallet in a single transaction, I had to first create a transaction sending most of my balance but leaving a small amount as change and then spend that change in a second transaction. This time it let me send all without creating an extra 0btc output.
I tried this on 2 separate HD wallets with multiple inputs and I had the same problem on both.

**Comment by ecdsa** [comment]
can you post a screenshot?

**Comment by niser32** [comment]
![electrum1](https://cloud.githubusercontent.com/assets/20031655/16259545/b03af318-3852-11e6-9706-e372943af021.png)
This is from a screenshot I took the day I posted this issue. Today I tried to send this exact same transaction but this time there was no change output. I couldn't replicate the problem on any wallet, looks like it just fixed itself.
Restarting Electrum didn't fix it because I did that before posting and the problem continued. I didn't reinstall the program, the only thing that I did since then is turning my computer off and on again, maybe that fixed it?
Sorry if I'm wasting your time, maybe this is some weird bug that only happens once in a century but it was very annoying for me then.
